src/utils/priceFetch.ts

```diff
@@ -1,22 +1,34 @@
 import axios from 'axios';
 import { getSentryTracker } from '../init';
 
+const FETCH_WINDOW_IN_MS = 60 * 1000;
+const ATTEMPTS = 5;
+
+let price = 0;
+let timestampInMS = 0;
+
 export const geckoEthToUsd = async (apiKey: string): Promise<number> => {
-  const attempts = 5;
+  const currentTimestampInMS = Date.now().valueOf();
+  if (currentTimestampInMS <= timestampInMS + FETCH_WINDOW_IN_MS) {
+    return price;
+  }
 
-  for (let attempt = 0; attempt < attempts; attempt += 1) {
+  price = 0;
+  for (let attempt = 0; attempt < ATTEMPTS; attempt += 1) {
     try {
       const data = await axios.get(
         `https://pro-api.coingecko.com/api/v3/simple/price?x_cg_pro_api_key=${apiKey}&ids=ethereum&vs_currencies=usd`,
       );
-      return data.data.ethereum.usd;
+      price = data.data.ethereum.usd;
```

Comment:
Previously the code stops if we manage to fetch eth price on first try due to the return. This code will fetch the eth five times in a row even though you get the eth price on the first go.

I think you introduce a small bug.

Reply:
Great spot!